libcloudforensics/providers/azure/internal/network.py

```
@@ -0,0 +1,180 @@
+  # the following cyclic import, as it it only used for type hints
+  from libcloudforensics.providers.azure.internal import account  # pylint: disable=cyclic-import
+
+
+class AZNetwork:
+  """Azure Networking functionality.
+
+  Attributes:
+    az_account (AZAccount): An Azure account object.
+    network_client (NetworkManagementClient): An Azure network client object.
+  """
+
+  def __init__(self,
+               az_account: 'account.AZAccount') -> None:
+    """Initialize the Azure network class.
+
+    Args:
+      az_account (AZAccount): An Azure account object.
+    """
+    self.az_account = az_account
+    self.network_client = network.NetworkManagementClient(
+        self.az_account.credentials, self.az_account.subscription_id)
+
+  def CreateNetworkInterface(self,
+                             name: str,
+                             region: Optional[str] = None) -> str:
+    """Create a network interface and returns its ID.
+
+    Args:
+      name (str): The name of the network interface.
+      region (str): Optional. The region in which to create the network
+          interface. Default uses default_region of the AZAccount object.
+
+    Returns:
+      str: The id of the created network interface.
+
+    Raises:
+      ValueError: if name is not provided.
+      RuntimeError: If no network interface could be created.
+    """
+    if not name:
+      raise ValueError('name must be specified. Provided: {0!s}'.format(name))
+
+    if not region:
+      region = self.az_account.default_region
+
+    network_interface_name = '{0:s}-nic'.format(name)
+    ip_config_name = '{0:s}-ipconfig'.format(name)
+
+    # Check if the network interface already exists, and returns its ID if so.
+    try:
+      nic = self.network_client.network_interfaces.get(
+          self.az_account.default_resource_group_name, network_interface_name)
+      nic_id = nic.id  # type: str
+      return nic_id
+    except azure_exceptions.CloudError:
```

Is it not possible that CloudError would also encapsulate other error situations than just "the NIC already exists"?
If something else goes wrong, we just eat the exception and continue..

---

Changed to raise the exception if it's not about the NIC not being found (which is the one we wanna ignore)